x-pack/plugins/security_solution/server/lib/detection_engine/rules/utils.ts

```diff
@@ -213,24 +216,61 @@ export const transformToAlertThrottle = (throttle: string | null | undefined): s
   }
 };
 
+/**
+ * Given a set of actions from an "alerting" Saved Object (SO) this will transform it into a "security_solution" alert action.
+ * If this detects any legacy rule actions it will transform it. If both are sent in which is not typical but possible due to
+ * the split nature of the API's this will prefer the usage of the non-legacy version. Eventually the "legacyRuleActions" should
+ * be removed.
+ * @param alertAction The alert action form a "alerting" Saved Object (SO).
+ * @param legacyRuleActions Legacy "side car" rule actions that if it detects it being passed it in will transform using it.
+ * @returns The actions of the FullResponseSchema
+ */
+export const transformActions = (
+  alertAction: AlertAction[] | undefined,
+  legacyRuleActions: LegacyRuleActions | null | undefined
+): FullResponseSchema['actions'] => {
+  if (alertAction != null && alertAction.length !== 0) {
+    return alertAction.map((action) => ({
+      group: action.group,
+      id: action.id,
+      action_type_id: action.actionTypeId,
+      params: action.params,
+    }));
+  } else if (legacyRuleActions != null) {
+    return legacyRuleActions.actions;
+  } else {
+    return [];
+  }
+};
+
 /**
  * Given a throttle from an "alerting" Saved Object (SO) this will transform it into a "security_solution"
- * throttle type.
- * @params throttle The throttle from a  "alerting" Saved Object (SO)
+ * throttle type. If given the "legacyRuleActions" but we detect that the rule for an unknown reason has actions
+ * on it to which should not be typical but possible due to the split nature of the API's, this will prefer the
+ * usage of the non-legacy version. Eventually the "legacyRuleActions" should be removed.
+ * @param throttle The throttle from a  "alerting" Saved Object (SO)
+ * @param legacyRuleActions Legacy "side car" rule actions that if it detects it being passed it in will transform using it.
  * @returns The "security_solution" throttle
  */
-export const transformFromAlertThrottle = (rule: SanitizedAlert<RuleParams>): string => {
-  if (rule.muteAll || rule.actions.length === 0) {
-    return NOTIFICATION_THROTTLE_NO_ACTIONS;
-  } else if (
-    rule.notifyWhen === 'onActiveAlert' ||
-    (rule.throttle == null && rule.notifyWhen == null)
-  ) {
-    return NOTIFICATION_THROTTLE_RULE;
-  } else if (rule.throttle == null) {
-    return NOTIFICATION_THROTTLE_NO_ACTIONS;
+export const transformFromAlertThrottle = (
+  rule: SanitizedAlert<RuleParams>,
+  legacyRuleActions: LegacyRuleActions | null | undefined
+): string => {
+  if (legacyRuleActions == null || (rule.actions != null && rule.actions.length > 0)) {
```

Comment:
NOTE: Double check my logic

Reply:
Frank, it looks like unit tests around this condition aren't there. Can you add some that exercises these new options being added. Probably would help with the code review as well.